### PR TITLE
New project structure

### PR DESCRIPTION
**Pull request by bullet-tooth**
## Overview

---
This is a base branch that represents the next structure:
```
exonum-java-parent
|
 - exonum-java-binding-parent
    |
     - exonum-java-binding-common
   .....
```
It will be used as a base branch to merge into small separated changes. 

### Definition of Done

- [ ] There is no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding-parent/exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ListSpliterator.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
  * associated with a given {@link com.exonum.binding.storage.database.Fork} is modified; not only
  * the source list.
  *
- * <p>This spliterator does not support specializations (e.g., {@link Spliterator.OfInt}).
+ * <p>This spliterator does not support specializations (e.g., {@link OfInt}).
```
Comment (dmitry-timofeev) [inline exonum-java-binding-parent/exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ListSpliterator.java]: It is not clear in the text what `OfInt` is.

File: exonum-java-binding-parent/pom.xml [hunk 2]
```diff
@@ -0,0 +1,647 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dmitry-timofeev) [inline exonum-java-binding-parent/pom.xml]: Git kind of breaks in such cases of moving and replacing original file, showing this one as brand new :)

File: pom.xml [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
-  <artifactId>exonum-java-binding-parent</artifactId>
+  <artifactId>exonum-java-parent</artifactId>
   <version>0.4-SNAPSHOT</version>
   <packaging>pom</packaging>
 
-  <name>Exonum Java Binding Parent</name>
+  <name>Exonum Java Parent</name>
   <description>
-    The Exonum Java Binding library.
+    The Exonum Java library.
```
Comment (dmitry-timofeev) [inline pom.xml]: The parent project of the Exonum Java libraries?

File: exonum-java-binding-parent/exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ListSpliterator.java (hunk 1 above)
Comment (bullet-tooth) [reply exonum-java-binding-parent/exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ListSpliterator.java]: Idea optimized it while merging with master :)

File: pom.xml [hunk 4]
```diff
@@ -70,577 +64,9 @@
...
-      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
-    </snapshotRepository>
-    <repository>
-      <id>ossrh</id>
-      <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
-    </repository>
-  </distributionManagement>
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (MakarovS) [inline pom.xml]: Do we need these two properties in both poms?

**Comment by bullet-tooth** [comment]
@dmitry-timofeev in #680

File: .travis.yml [hunk 5]
```diff
@@ -9,6 +9,8 @@ branches:
     - ejb-app-configuration
```
Comment (bullet-tooth) [inline .travis.yml]: is it still needed?

File: exonum-java-binding/core/rust/integration_tests/src/vm.rs [hunk 6]
```diff
@@ -102,15 +102,15 @@ fn get_fakes_classpath_option() -> String {
 
 pub fn get_fakes_classpath() -> String {
     let classpath_txt_path =
-        project_root_dir().join("exonum-java-binding-fakes/target/ejb-fakes-classpath.txt");
+        project_root_dir().join("fakes/target/ejb-fakes-classpath.txt");
```
Comment (bullet-tooth) [inline exonum-java-binding/core/rust/integration_tests/src/vm.rs]: @skletsun @vitvakatu please check

**Pull request merged**